apps/cowswap-frontend/src/modules/hooksStore/pure/AddCustomHookForm/CustomDappLoader/index.tsx

```diff
@@ -54,23 +97,37 @@ export function ExternalDappLoader({
             url: input,
           })
         }
-      })
-      .catch((error) => {
-        if (!isRequestRelevant) return
-
-        console.error(error)
-        setManifestError('Can not fetch the manifest.json')
-      })
-      .finally(() => {
-        if (!isRequestRelevant) return
+      } catch (error) {
+        console.error('Hook dapp loading error:', error)
 
+        if (error.message?.includes('JSON')) {
+          setManifestError(ERROR_MESSAGES.INVALID_MANIFEST_HTML)
+        } else if (error.name === 'AbortError') {
```

Comment:
This check is done twice. Ideally the string is moved to a const, and the check itself to a utility function such us `isTimeoutError(error: unknown)`

Reply:
Addressed this now